### DISCUSSION TITLE
[BUG] [Issue] DAILY GET_LEDGER_SUMMARY_VIEW_DATA report

### DISCUSSION THREAD

**Discussion by yashraj021** [amzn/selling-partner-api-models#4179] [Q&A]
Hi,

I am trying to download Ledger report using AP-API but getting empty rows. However, I am able to download it from India Seller Central dashboard. 

Below attaching payload and screenshot of API calls:

I have also used date dormat without adding 05:30 in format, i.e, and still getting zero rows.

    "dataStartTime": "2024-09-14T00:00:00Z",
    "dataEndTime": "2024-09-15T23:59:59Z",

My payload:

> {
    "marketplaceIds": [
        "A21TJRUUN4KGV"
    ],
    "reportType": "GET_LEDGER_SUMMARY_VIEW_DATA",
    "dataStartTime": "2024-09-14T00:00:00+05:30",
    "dataEndTime": "2024-09-15T23:59:59+05:30",
    "reportOptions":{
        "aggregateByLocation":"FC",
        "aggregateByTimePeriod":"DAILY"
    }
}

Screenshot Below:
![Screenshot 2024-09-16 at 11 22 35 AM](https://github.com/user-attachments/assets/7ea10362-507e-467f-9832-88aea21dcebf)

**Comment by jawheral** [comment]
Hi @yashraj021,

The problem seems to be originating from the `dateStartTime`. Please refer to [Inventory Ledger Report - Summary View](https://developer-docs.amazon.com/sp-api/docs/report-type-values-fba#inventory-ledger-report---summary-view).

Note the "*If aggregatedByTimePeriod is set as DAILY, Ledger Summary View will return data aggregated at the end of day. For example, dataStartTime is 2023-01-15T00:00:00+00:00 and dataEndTime is 2023-01-15T23:59:59+00:00 will return Jan 15th data*" and  "*Ledger Report Summary View can return empty data if invalid dataStartTime/dataEndTime is passed along with aggregatedByTimePeriod filter values*".

This means `dateStartTime` should be on the 15th: `2024-09-15T00:00:00Z` or `2024-09-15T00:00:00+05:30` both are correct Datetime values but depends on the data you need.

I hope this helps!

Best,
Jawher
Selling Partner API Solutions Architect

**Reply by yashraj021** [reply]
> Hi @yashraj021,
> 
> The problem seems to be originating from the `dateStartTime`. Please refer to [Inventory Ledger Report - Summary View](https://developer-docs.amazon.com/sp-api/docs/report-type-values-fba#inventory-ledger-report---summary-view).
> 
> Note the "_If aggregatedByTimePeriod is set as DAILY, Ledger Summary View will return data aggregated at the end of day. For example, dataStartTime is 2023-01-15T00:00:00+00:00 and dataEndTime is 2023-01-15T23:59:59+00:00 will return Jan 15th data_" and "_Ledger Report Summary View can return empty data if invalid dataStartTime/dataEndTime is passed along with aggregatedByTimePeriod filter values_".
> 
> This means `dateStartTime` should be on the 15th: `2024-09-15T00:00:00Z` or `2024-09-15T00:00:00+05:30` both are correct Datetime values but depends on the data you need.
> 
> I hope this helps!
> 
> Best, Jawher Selling Partner API Solutions Architect

Hi @jawheral ,

First of all thank you for your quick response. Highly appreciated!

Secondly, I have tried 2024-09-15T00:00:00Z to 2024-09-15T23:59:59Z and still it didnt produce any rows.

Can you tell me in what other cases it wont produce rows?

**Reply by jawheral** [reply]
Hi @yashraj021,

What happens when you request some other past days. For example 12th or 13th? Do you also receive empty data?

**Reply by yashraj021** [reply]
> Hi @yashraj021,
> 
> What happens when you request some other past days. For example 12th or 13th? Do you also receive empty data?

I have requested for below date time configuration on **Amazon Seller Portal** and getting the rows for **FC**. However doing same with API is not working.
"dataStartTime": "2024-09-14T00:00:00+05:30",
"dataEndTime": "2024-09-15T23:59:59+05:30",

But I never get any data if I have put the date difference of max 24 hrs, for example:
"dataStartTime": "2024-09-12T00:00:00+05:30",
"dataEndTime": "2024-09-12T23:59:59+05:30",

or

"dataStartTime": "2024-09-12T00:00:00Z",
"dataEndTime": "2024-09-12T23:59:59Z",

**Reply by jawheral** [reply]
Did you make sure that you are calling on behalf of the right seller and that your developer profile and app have the Amazon Fulfillment role checked?

**Reply by yashraj021** [reply]
> Did you make sure that you are calling on behalf of the right seller and that your developer profile and app have the Amazon Fulfillment role checked?

Yes. I have been using these api's for multiple reports like: `GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE_V2` and receiving data properly.

Also in case of LEDGER reports, I am actually able to get data for monthly basis if I mention start date of March 2024, but not if I select DAILY and mention a date (example below). 

"dataStartTime": "2024-03-12T00:00:00Z",
"dataEndTime": "2024-03-12T23:59:59Z",

And, i am getting data if I try to download from Seller Central portal with any configuration.

**Reply by jawheral** [reply]
@yashraj021 In this case, please open a case with our [Developer Support](https://developer.amazonservices.com/support) while providing all the needed details such example requestId or reportId that bring empty reports but should include data.

Best,
Jawher
Selling Partner API Solutions Architect

**Reply by yashraj021** [reply]
> @yashraj021 In this case, please open a case with our [Developer Support](https://developer.amazonservices.com/support) while providing all the needed details such example requestId or reportId that bring empty reports but should include data.
> 
> Best, Jawher Selling Partner API Solutions Architect

Thank you very much for your assistance. I will raise the ticket.
